bot/cogs/moderation/incidents.py

```diff
@@ -186,22 +186,41 @@ async def archive(self, incident: discord.Message, outcome: Signal, actioned_by:
 
         The following pieces of information are relayed:
             * Incident message content (as embed description)
+            * Incident attachment (if image, shown in archive embed)
             * Incident author name (as webhook author)
             * Incident author avatar (as webhook avatar)
             * Resolution signal `outcome` (as embed colour & footer)
             * Moderator `actioned_by` (name & discriminator shown in footer)
 
+        If `incident` contains an attachment, we try to add it to the archive embed. There is
+        no handing of extensions / file types - we simply dispatch the attachment file with the
+        webhook, and try to display it in the embed. Testing indicates that if the attachment
+        cannot be displayed (e.g. a text file), it's invisible in the embed, with no error.
+
         Return True if the relay finishes successfully. If anything goes wrong, meaning
         not all information was relayed, return False. This signals that the original
         message is not safe to be deleted, as we will lose some information.
         """
         log.debug(f"Archiving incident: {incident.id} (outcome: {outcome}, actioned by: {actioned_by})")
+        embed = make_embed(incident, outcome, actioned_by)
+
+        # If the incident had an attachment, we will try to relay it
+        if incident.attachments:
+            attachment = incident.attachments[0]  # User-sent messages can only contain one attachment
+            log.debug(f"Attempting to archive incident attachment: {attachment.filename}")
+
+            attachment_file = await attachment.to_file()  # The file will be sent with the webhook
```

Comment:
Good point, this should be addressed. But I'm already anxious thinking about how to write this so that it's not too ugly. I'll ponder upon it.

Reply:
Actually I wonder if this should abort entirely? If this fails for some reason with the attachment being valid & important but we still relay, we will lose potentially important information.

Reply:
You can fall back to linking the attachment or add a note that there was supposed to be one but it failed.

Reply:
How would you feel about using the author name field?

![image](https://user-images.githubusercontent.com/44734341/87154004-be422b80-c2b8-11ea-9fa3-ae41859fd567.png)

It can be made to link the attachment proxy url, but I think that stops being functional some time after the original message is deleted.

Reply:
My first though would be to put it at the end of the description so that the message content "takes priority". However, I am fine with your proposal as well.

> I do log the exception so we'll get a Sentry notif if this happens.

I don't think `NotFound` and any 500's are worth being notified for, though we should probably consider ignoring all 500s at a larger scale rather than here.

Reply:
I preferred the name field because (although it's really not a name) it prevents having to worry about the embed description length, plus I think it looks ok.

I'm still a little bit on the edge though, I feel like it may be more logical to just return if the download fails. I'm considering moving the logic elsewhere, the `archive` function is now branching too much and I'm struggling to adjust the tests. :angry:

Reply:
>  I feel like it may be more logical to just return if the download fails. 

Yes, if the message has only attachments and no text content. Otherwise, why?

Reply:
Take a peek at c4e9060 and 6e48d66.

I added a method to download the file safely, but it currently error-level logs all exceptions. Additional clauses can be put there to reduce the log level of specific exceptions (maybe the 404? but in practice it really shouldn't happen unless someone deleted the msg manually as it's being relayed).

The latter commit then moves the attachment-handling logic to `make_embed`, I decided to handle it all there and return the file if available. Seems more logical in the end. Keeps `archive` simple.

Let me know how you feel about this.